x/ccv/consumer/keeper/params.go

```diff
@@ -32,6 +34,14 @@ func (k Keeper) SetParams(ctx sdk.Context, params types.Params) {
 	k.paramStore.SetParamSet(ctx, &params)
 }
 
+// GetParams implements StakingKeeper GetParams interface method
+// it returns an a empty stakingtypes.Params struct
+// NOTE: this method must be implemented on the consumer keeper because the evidence module keeper
+// in cosmos-sdk v0.47 requires this exact method with this exact signature to be available on the StakingKeepr
+func (k Keeper) GetParams(ctx sdk.Context) stakingtypes.Params {
```

Comment:
@mpoke 
This is required for Slashing (IIRC). Would it be safe to populate the `stakingtypes.Prams` with the corresponding params in `types.Params`?

`types.Params` contain all `stakingtypes.Params` fields:
```golang
type Params struct {
	// unbonding_time is the time duration of unbonding.
	UnbondingTime time.Duration `protobuf:"bytes,1,opt,name=unbonding_time,json=unbondingTime,proto3,stdduration" json:"unbonding_time"`
	// max_validators is the maximum number of validators.
	MaxValidators uint32 `protobuf:"varint,2,opt,name=max_validators,json=maxValidators,proto3" json:"max_validators,omitempty"`
	// max_entries is the max entries for either unbonding delegation or redelegation (per pair/trio).
	MaxEntries uint32 `protobuf:"varint,3,opt,name=max_entries,json=maxEntries,proto3" json:"max_entries,omitempty"`
	// historical_entries is the number of historical entries to persist.
	HistoricalEntries uint32 `protobuf:"varint,4,opt,name=historical_entries,json=historicalEntries,proto3" json:"historical_entries,omitempty"`
	// bond_denom defines the bondable coin denomination.
	BondDenom string `protobuf:"bytes,5,opt,name=bond_denom,json=bondDenom,proto3" json:"bond_denom,omitempty"`
	// min_commission_rate is the chain-wide minimum commission rate that a validator can charge their delegators
	MinCommissionRate github_com_cosmos_cosmos_sdk_types.Dec `protobuf:"bytes,6,opt,name=min_commission_rate,json=minCommissionRate,proto3,customtype=github.com/cosmos/cosmos-sdk/types.Dec" json:"min_commission_rate" yaml:"min_commission_rate"`
}
```

Just remembered this and have a hunch that it may be important, but I didn't see anything break during testing.

Reply:
On second thought, this returns `MaxValidators` which we cannot populate.

I'll leave this here for posterity.

Reply:
> This is required for Slashing (IIRC).

It is actually required by the evidence module https://github.com/cosmos/cosmos-sdk/blob/666c345ad23ddda9523cc5cd1b71187d91c26f34/x/evidence/types/expected_keepers.go#L18, and it's only used for testing. See https://github.com/cosmos/cosmos-sdk/pull/12749. I think it's fine to leave it as a noop here.